packages/app-layout/AppLayout.tsx

```diff
@@ -0,0 +1,161 @@
+import { h, Component, prop } from 'skatejs';
+
+import {
+  GenericTypes,
+  matchMedia
+} from '../_helpers';
+
+import { HamburgerButton } from './components/HamburgerButton';
+import { Drawer } from './components/Drawer';
+import { Overlay } from './components/Overlay';
+import { Nav } from './components/Nav';
+import { NavContent } from './components/NavContent';
+
+import styles from './AppLayout.scss';
+
+const DEFAULT_RESPONSIBLE_WIDTH = '640px';
+
+type AppLayoutProps = Props & EventProps;
+
+type Attrs = {
+  'drawer-visible'?: boolean,
+  'responsive-width'?: string
+};
+
+type Props = {
+  drawerVisible?: boolean,
+  responsiveWidth?: string
+};
+
+// TODO: add onDrawerChange / onNarrowChange
+type EventProps = {};
+
+type Events = {};
+
+declare global {
+  namespace JSX {
+    interface IntrinsicElements {
+      'bl-app-layout': GenericTypes.IntrinsicCustomElement<AppLayoutProps>
+      & GenericTypes.IntrinsicBoreElement<Attrs, Events>
+    }
+  }
+}
+
+export class AppLayout extends Component<AppLayoutProps> {
+
+  static get is() { return 'bl-app-layout'; }
+
+  static get props() {
+    return {
+      drawerVisible: prop.boolean( {
+        attribute: {
+          source: true
+        }
+      } ),
+      responsiveWidth: prop.string( {
+        initial: DEFAULT_RESPONSIBLE_WIDTH,
+        attribute: {
+          source: true
+        }
+      } ),
+      narrow: prop.boolean( {
+        attribute: {
+          source: true,
```

Comment:
ah you are right I kind of thought that attribute: { source: true } will be default for all pops, but it will be default only for attribute: true, ok will change it to not use attribute reflection at all